backend/src/apiserver/template/v2_template.go

```diff
@@ -107,37 +110,72 @@ func (t *V2Spec) GetTemplateType() TemplateType {
 
 func NewV2SpecTemplate(template []byte) (*V2Spec, error) {
 	var spec pipelinespec.PipelineSpec
-	templateJson, err := yaml.YAMLToJSON(template)
-	if err != nil {
-		return nil, util.NewInvalidInputErrorWithDetails(ErrorInvalidPipelineSpec, fmt.Sprintf("cannot convert v2 pipeline spec to json format: %s", err.Error()))
-	}
-	err = protojson.Unmarshal(templateJson, &spec)
-	if err != nil {
-		return nil, util.NewInvalidInputErrorWithDetails(ErrorInvalidPipelineSpec, fmt.Sprintf("invalid v2 pipeline spec: %s", err.Error()))
-	}
-	if spec.GetPipelineInfo().GetName() == "" {
-		return nil, util.NewInvalidInputErrorWithDetails(ErrorInvalidPipelineSpec, "invalid v2 pipeline spec: name is empty")
-	}
-	match, _ := regexp.MatchString("[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*", spec.GetPipelineInfo().GetName())
-	if !match {
-		return nil, util.NewInvalidInputErrorWithDetails(ErrorInvalidPipelineSpec, "invalid v2 pipeline spec: name should consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character")
+	var v2Spec V2Spec
+	decoder := goyaml.NewDecoder(bytes.NewReader(template))
+	for {
+		var value map[string]interface{}
+		// Break at end of file
+		if decoder.Decode(&value) != nil {
+			break
+		}
+		valueBytes, err := goyaml.Marshal(value)
+		if err != nil {
+			return nil, util.NewInvalidInputErrorWithDetails(ErrorInvalidPipelineSpec, fmt.Sprintf("unable to marshal this yaml document: %s", err.Error()))
```

Comment:
I am probably wrong, but I expected something like this:
```golang
for _, templateSpec := strings.Split(string(template), "\n---\n") {
  if isPipelineSpec([]byte(templateSpec)){
      // <throw error if PipelineSpec has been already parse (duplicate PipelineSpec)>
      // <parse into *PipelineSpec>
  } else if isKubernetesExecutorConfig([]byte(templateSpec)){
      // <throw error if PlatformSpec has been already parse (duplicate PlatformSpec)>
      // <parse into *PlatformSpec>
  }
}
// <throw error if PipelineSpec has not been found in the input string>
```

Reply:
I think both ways can parse the input of multiple yaml files, but I think you have a good point that we should verify that both inputs are unique. I will change the code accordingly.

Reply:
done.

Reply:
Thank you adding the errors!

Could you also clarify why do we need to use:
```golang
decoder := goyaml.NewDecoder(bytes.NewReader(template))
for {
	var value map[string]interface{}
	// Break at end of file
	if decoder.Decode(&value) != nil {
		break
	}
...
}
```
instead of this:

```golang
for _, templateSpec := strings.Split(string(template), "\n---\n") {
      templateBytes := []byte(templateSpec)
...
}
```

Reply:
Thank you, @Linchin!